src/features/account/components/modals/AddToUserCircleModal.tsx

```diff
@@ -26,50 +28,75 @@ export const AddToUserCircleModal: FunctionComponent<
   );
   const dispatch = useDispatch();
 
-  const {
-    isLoading,
-    isVerified,
-    payload,
-    submitTypedSignRequest,
-    signatureError,
-  } = useTypedSignTrustCredential();
+  const [isLoading, setIsLoading] = useState(false);
+  const { issuerAddress, signMessage, signError, accountVCBuilder } =
+    useVerifiableCredential();
 
-  const { showErrorMsg, showSuccessMsg } = useToastMsg();
+  const { showSuccessMsg, showErrorMsg } = useToastMsg();
 
   const shortSubAddress = useMemo(
     () => trimAddress(subjectAddress),
     [subjectAddress],
   );
 
-  useEffect(() => {
-    if (signatureError) {
-      if (signatureError.type === SignatureErrorTypes.Error) {
-        showErrorMsg({
-          title: t`Failed to verify signature`,
-          description: signatureError.message,
-        });
-      } else {
+  const onSignButtonClick = useCallback(() => {
+    if (!issuerAddress) {
+      return;
+    }
+    if (!isAddress(subjectAddress)) {
+      return;
+    }
+    setIsLoading(true);
+
+    const VC = accountVCBuilder.buildAccountTrust(
+      issuerAddress,
+      subjectAddress,
+    );
+
+    signMessage(VC)
+      .then((signature) => {
+        if (signature) {
+          const assertion = accountVCBuilder.getSignedAssertion(VC, signature);
+          console.log('assertion', assertion);
+
+          showSuccessMsg({
+            title: t`Added to your trust circle`,
+            description: t`${shortSubAddress} has been added to your trust circle`,
+          });
+          dispatch(addUserToUserCircle(subjectAddress));
+          dispatch(setAddToUserModalOpen(false));
+        } else {
+          showErrorMsg({
+            title: t`Error`,
+            description: t`The signature is empty`,
+          });
+        }
+      })
+      .catch((error) => {
+        console.error(error);
         showErrorMsg({
-          title: t`Invalid Signature`,
-          description: t`Your signature is invalid`,
+          title: t`Error`,
+          description: t`Unknown error`,
         });
-      }
```

Comment:
you dont need to catch it, underlying handled and use signError to return

Reply:
without catch, the lint will failed mentioned promise without catch error here. but if you change promise to use async way here, lint will still failed complaining `signMessage` return null, promise<void>.

The extra code here just to prevent the lint error.

Reply:
i think you can use await

Reply:
here is the example without lint error
![image](https://github.com/MetaMask/permissionless-snaps-directory/assets/102275989/615d058c-4a7c-45a7-be1d-72573d432845)

Reply:
Type error will throw here, let me show you what i mean.

Reply:
After discussion and try a lot of way, the latest change can use finally to avoid the lint error to solve this issue.